src/components/MAPI/utils.ts

```diff
@@ -348,6 +348,37 @@ export function fetchClusterKey(
   return capiv1alpha3.getClusterKey(namespace, name);
 }
 
+export interface IClusterListWithProviderClusters {
+  clusterList: ClusterList;
+  providerClusters: IProviderClusterForClusterName[];
+}
+
+export async function fetchClusterListWithProviderClusters(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  provider: PropertiesOf<typeof Providers>,
+  namespace?: string,
+  organization?: string
+): Promise<IClusterListWithProviderClusters> {
+  const clusterList = await fetchClusterList(
+    httpClientFactory,
+    auth,
+    provider,
+    namespace,
+    organization
+  );
```

Comment:
Small suggestion here - since we call this function in other places in the app, maybe we could leverage `swr`'s cache here as well?

Reply:
good point, but not actual anymore since I returned to the previous logic to fetch cluster list and provider clusters separately.